# Actividad 2
## Carga de Base de Datos

In [8]:
import pandas as pd
df = pd.read_csv('../Laboratorio_A01706695/archive/cwurData.csv')
df

world_rank                            institution         country  \
0              1                     Harvard University             USA   
1              2  Massachusetts Institute of Technology             USA   
2              3                    Stanford University             USA   
3              4                University of Cambridge  United Kingdom   
4              5     California Institute of Technology             USA   
...          ...                                    ...             ...   
2195         996              University of the Algarve        Portugal   
2196         997                  Alexandria University           Egypt   
2197         998            Federal University of Ceará          Brazil   
2198         999                 University of A Coruña           Spain   
2199        1000        China Pharmaceutical University           China   

      national_rank  quality_of_education  alumni_employment  \
0                 1                     7                  9   
1                 2                     9                 17   
2                 3                    17                 11   
3                 1                    10                 24   
4                 4                     2                 29   
...             ...                   ...                ...   
2195              7                   367                567   
2196              4                   236                566   
2197             18                   367                549   
2198             40                   367                567   
2199             83                   367                567   

      quality_of_faculty  publications  influence  citations  broad_impact  \
0                      1             1          1          1           NaN   
1                      3            12          4          4           NaN   
2                      5             4          2          2           NaN   
3                      4            16         16         11           NaN   
4                      7            37         22         22           NaN   
...                  ...           ...        ...        ...           ...   
2195                 218           926        845        812         969.0   
2196                 218           997        908        645         981.0   
2197                 218           830        823        812         975.0   
2198                 218           886        974        812         975.0   
2199                 218           861        991        812         981.0   

      patents   score  year  
0           5  100.00  2012  
1           1   91.67  2012  
2          15   89.50  2012  
3          50   86.17  2012  
4          18   85.21  2012  
...       ...     ...   ...  
2195      816   44.03  2015  
2196      871   44.03  2015  
2197      824   44.03  2015  
2198      651   44.02  2015  
2199      547   44.02  2015  

[2200 rows x 14 columns]

## Exploración de datos

### Uso de .info()

In [9]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2200 entries, 0 to 2199
Data columns (total 14 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   world_rank            2200 non-null   int64  
 1   institution           2200 non-null   object 
 2   country               2200 non-null   object 
 3   national_rank         2200 non-null   int64  
 4   quality_of_education  2200 non-null   int64  
 5   alumni_employment     2200 non-null   int64  
 6   quality_of_faculty    2200 non-null   int64  
 7   publications          2200 non-null   int64  
 8   influence             2200 non-null   int64  
 9   citations             2200 non-null   int64  
 10  broad_impact          2000 non-null   float64
 11  patents               2200 non-null   int64  
 12  score                 2200 non-null   float64
 13  year                  2200 non-null   int64  
dtypes: float64(2), int64(10), object(2)
memory usage: 240.8+ KB


Gracias a esta función, se pueden observar las columnas, la cantidad y el tipo de datos que nuestra base de datos posee.   
Este análisis muestra que la mayoría de los datos presentes son de tipo **entero**; tambien se poseen dos datos del tipo **objeto** para las columnas de *institution* y *country*, finalmente, exiten también dos tipos **flotantes** para las categorías *score* y *broad_impact* respectivamete.

### Uso de .count()

Lo siguiente muestra la cantidad de valores que hay para comprobar si hay *valores vacíos*.

In [10]:
df.count()

world_rank              2200
institution             2200
country                 2200
national_rank           2200
quality_of_education    2200
alumni_employment       2200
quality_of_faculty      2200
publications            2200
influence               2200
citations               2200
broad_impact            2000
patents                 2200
score                   2200
year                    2200
dtype: int64

Utilizar esta herramienta confirma que gran parte de los datos están completos a excepción de algunos en la sección *broad_impact* como para ralizar el análisis estadístico no son importantes, esto no debería de representar un problema.

### Uso de len()

In [11]:
len(df)

2200

Con esta función también se observa la cantidad de datos que se poseen.

## Análisis Estadístico

Para esta parte, este análisis se llevará a cabo utilizando como referencia la columna de *score*; para obtener otros datos, esta también puede utilizarse en relación con otras columnas.

## Media
Como primer operación a realizar, se obtendrá el promedio de la columna anteriormente mencionada

In [12]:
print('Promedio/Media: ', df['score'].mean())

Promedio/Media:  47.79839545454546


La calificación promedio que hay entre estas universidades es de *47.8* aproximadamente.
Para saber más acerca de países que están entre el 40 y 50 de calificación, se hace uso de un **filtro**:

In [13]:
enPromedio = df[(df['score'] >= 40) & (df['score'] <= 50)]
enPromedio

world_rank                       institution   country  national_rank  \
51            52   Ohio State University, Columbus       USA             38   
52            53           University of Rochester       USA             39   
53            54  École normale supérieure - Paris    France              2   
54            55             University of Florida       USA             40   
55            56               Tel Aviv University    Israel              4   
...          ...                               ...       ...            ...   
2195         996         University of the Algarve  Portugal              7   
2196         997             Alexandria University     Egypt              4   
2197         998       Federal University of Ceará    Brazil             18   
2198         999            University of A Coruña     Spain             40   
2199        1000   China Pharmaceutical University     China             83   

      quality_of_education  alumni_employment  quality_of_faculty  \
51                     101                101                  83   
52                      41                 54                 101   
53                       3                 61                  77   
54                     101                101                  96   
55                      97                101                  51   
...                    ...                ...                 ...   
2195                   367                567                 218   
2196                   236                566                 218   
2197                   367                549                 218   
2198                   367                567                 218   
2199                   367                567                 218   

      publications  influence  citations  broad_impact  patents  score  year  
51              30         50         35           NaN       54  49.54  2012  
52              82         68         54           NaN       62  49.51  2012  
53             101        101        101           NaN      101  49.10  2012  
54              29         74         42           NaN        2  48.80  2012  
55              72        101        101           NaN       35  48.76  2012  
...            ...        ...        ...           ...      ...    ...   ...  
2195           926        845        812         969.0      816  44.03  2015  
2196           997        908        645         981.0      871  44.03  2015  
2197           830        823        812         975.0      824  44.03  2015  
2198           886        974        812         975.0      651  44.02  2015  
2199           861        991        812         981.0      547  44.02  2015  

[1868 rows x 14 columns]

In [14]:
len(enPromedio)

1868

En conjunto con la función *len*, se puede conocer que ***1868*** universidades de las ***2200*** están en el rango intermedio.